### ISSUE TITLE
Stylelint throws CssSyntaxError on JSON files

### ISSUE THREAD

**Issue by geekox86** [stylelint/stylelint#4075]
> Clearly describe the bug

Stylelint includes JSON files for linting and throws CssSyntaxError.

> Which rule, if any, is the bug related to?

This bug is not related to any specific rule.

> What CSS is needed to reproduce the bug?

The existent of `package.json` and `tsconfig.json` in root project directory can reproduce the bug.

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": ["stylelint-config-standard"],
  "rules": {}
}
```

> Which version of stylelint are you using?

`10.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Using CLI with `yarn stylelint .`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

I expected Stylelint *not* to include JSON files as it contains no styles.

> What actually happened (e.g. what warnings or errors did you get)?

```shell
/usr/local/bin/node /usr/local/Cellar/yarn/1.15.2/libexec/bin/yarn.js run lint:styles
yarn run v1.15.2
$ yarn stylelint .
$ /Users/geekox86/Documents/GitHub/vue-apollo-graphql/node_modules/.bin/stylelint .

package.json
 11:101  ✖  Missed semicolon   CssSyntaxError

tsconfig.json
 15:18  ✖  Unknown word   CssSyntaxError

codes/index.css
 7:1  ✖  Unexpected missing end-of-source newline   no-missing-end-of-source-newline

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Process finished with exit code 2
```

Here is a [reproduction repo](https://github.com/geekox86/vue-apollo-graphql) for your testing.

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

stylelint doesn't support `stylelint .` at the moment, but we [want](https://github.com/stylelint/stylelint/issues/3411) to support it. Currently, you have to specify all extensions. For your project you might want to use:

```
stylelint 'codes/**/*.css'
```

**Comment by evilebottnawi** [comment]
@hudochenkov i think we can close this in favor https://github.com/stylelint/stylelint/issues/3411

**Comment by LcsGrz** [comment]
> Thanks for the report and for using the template.
> 
> stylelint doesn't support `stylelint .` at the moment, but we [want](https://github.com/stylelint/stylelint/issues/3411) to support it. Currently, you have to specify all extensions. For your project you might want to use:
> 
> ```
> stylelint 'codes/**/*.css'
> ```

Hi, I hope you're well, i have the same problem with same files, but from the ide

<img width="1054" alt="imagen" src="https://github.com/stylelint/stylelint/assets/21349409/cee37427-3c65-42ff-a4ec-f4885076e19f">

what is the real problem?
